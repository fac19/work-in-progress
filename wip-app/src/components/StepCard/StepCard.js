import React from "react";
import { StepCardArticle, StepCardImage } from "./StepCard.style";
import { useHistory } from "react-router-dom";

const StepCard = ({ id, step_name, step_description, step_link }) => {
  const history = useHistory();
  const [stepData, setStepData] = React.useState(null);

  const goToStepPage = () => {
    history.push(`/step/${id}`);
  };

  return (
    <StepCardArticle onClick={goToStepPage}>
      <h3>{step_name}</h3>
      <p>{step_description}</p>
      <StepCardImage src={step_link} alt="" />
    </StepCardArticle>
  );
};

export default StepCard;
