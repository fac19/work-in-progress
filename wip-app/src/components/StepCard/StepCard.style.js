import styled from "styled-components";

const StepCardArticle = styled.article`
  margin: 2rem;
  padding: 2rem;
  border-radius: 2px;
  box-shadow: 0px 0px 10px grey;
  text-align: center;
`;

const StepCardImage = styled.img`
  width: 100%;
`;

export { StepCardArticle, StepCardImage };
