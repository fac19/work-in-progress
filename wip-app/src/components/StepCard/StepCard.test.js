import React from "react";
import TestRenderer from "react-test-renderer";
import StepCard from "./StepCard";

describe("Snapshot test", () => {
  it("renders the correct ui", () => {
    const component = TestRenderer.create(<StepCard />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
