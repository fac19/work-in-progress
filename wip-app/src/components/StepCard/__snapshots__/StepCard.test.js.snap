// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test renders the correct ui 1`] = `
<article
  className="sc-AxjAm jQidqc"
  onClick={[Function]}
>
  <h3 />
  <p />
  <img
    alt=""
    className="sc-AxirZ cUVYWJ"
  />
</article>
`;
